cookbook/examples/apps/image_generation/agents.py

```diff
@@ -0,0 +1,75 @@
+# Recipe agent for image generation
+from textwrap import dedent
+from typing import Optional
+
+from agno.agent import Agent
+from agno.document.reader.pdf_reader import PDFImageReader
+from agno.embedder.cohere import CohereEmbedder
+from agno.knowledge.pdf import PDFKnowledgeBase
+from agno.knowledge.pdf_url import PDFUrlKnowledgeBase
+from agno.models.groq import Groq
+from agno.tools.openai import OpenAITools
+from agno.vectordb.pgvector import PgVector
+
+# Database connection string for recipe knowledge base
+# Adjust as needed for your environment
+DB_URL = "postgresql+psycopg://ai:ai@localhost:5532/ai"
+
+# Constants for recipe agent
+DEFAULT_RECIPE_URL = "https://agno-public.s3.amazonaws.com/recipes/ThaiRecipes.pdf"
```

Comment:
Lets add a few more recipe books to make it more compelling? Just download random ones and upload to the S3 bucket (I can also help)